drivers/modem/modem_cellular.c

```diff
@@ -293,7 +317,48 @@ static void modem_cellular_chat_on_cgmm(struct modem_chat *chat, char **argv, ui
 		return;
 	}
 
-	strncpy(data->hwinfo, argv[1], sizeof(data->hwinfo) - 1);
+	strncpy(data->fw_version, argv[1], sizeof(data->fw_version) - 1);
+}
+
+static void modem_cellular_chat_on_csq(struct modem_chat *chat, char **argv, uint16_t argc,
+				       void *user_data)
+{
+	/* AT+CSQ returns a response +CSQ: <rssi>,<ber> where:
+	 * - rssi is a integer from 0 to 31 whose values describes a signal strength
+	 *   between -113 dBm for 0 and -51dbM for 31 or unknown for 99
+	 * - ber is an integer from 0 to 7 that describes the error rate, it can also
+	 *   be 99 for an unknown error rate
+	 */
+	struct modem_cellular_data *data = (struct modem_cellular_data *)user_data;
+
+	if (argc != 3) {
+		return;
+	}
+
+	/* Read rssi */
+	uint8_t rssi = atoi(argv[1]);
+
+	if (rssi == 99) {
+		return;
+	}
+
+	data->rssi = (-113 + (2 * rssi));
+}
+
+static void modem_cellular_chat_on_imsi(struct modem_chat *chat, char **argv, uint16_t argc,
+					void *user_data)
+{
+	struct modem_cellular_data *data = (struct modem_cellular_data *)user_data;
+
+	strncpy(data->imsi, (char *)argv[1], sizeof(data->imsi));
```

Comment:
Perhaps I'm entirely wrong, but aren't the cases of IMSI and ICCID the same as the rest? Why do we have a different treatment? There is no need to cast argv[1] to char * (it already is), and we need to put again sizeof(data->imsi) - 1. The same with ICCID. Right?

Reply:
On second thoughts, I actually think that those two are correct and most probably the imei is wrong. We have assigned the exact amount of bytes in the array, so we have to use `sizeof(data->imei)` and not `sizeof(data->imei) - 1`